ftw/zipextract/configure.zcml

```diff
@@ -0,0 +1,50 @@
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:genericsetup="http://namespaces.zope.org/genericsetup"
+    xmlns:five="http://namespaces.zope.org/five"
+    xmlns:upgrade-step="http://namespaces.zope.org/ftw.upgrade"
+    xmlns:zcml="http://namespaces.zope.org/zcml"
+    i18n_domain="ftw.zipextract">
+
+    <five:registerPackage package="." initialize=".initialize" />
+
+    <genericsetup:registerProfile
+        name="default"
+        title="ftw.zipextract default"
+        directory="profiles/default"
+        provides="Products.GenericSetup.interfaces.EXTENSION"
+        />
+
+    <genericsetup:registerProfile
+        name="uninstall"
+        title="ftw.zipextract : uninstall"
+        directory="profiles/uninstall"
+        provides="Products.GenericSetup.interfaces.EXTENSION"
+        for="Products.CMFPlone.interfaces.IMigratingPloneSiteRoot"
+        />
+
+    <include package=".browser" />
+    <include package="ftw.upgrade" file="meta.zcml" />
+    <upgrade-step:directory
+        profile="ftw.zipextract:default"
+        directory="upgrades"
+        />
+
+    <configure zcml:condition="installed ftw.file">
+        <adapter
+            for="ftw.file.content.file.IFile"
```

Comment:
I'd try to implement a "archetypes" default implementation. I think it should be possibl to not make it `ftw.file` specific while still making it work with `ftw.file`.
We can rely on the file-field beeing defined as primary field, so that we can do thinks like `context.getPrimaryField().getBlob()` and such.

If it needs to be `ftw.file`-specific, it belongs in the `ftw.file`-package. In this case the `ftw.file`-package declares a `tests`-dependency to `ftw.zipextract`.

Reply:
Done. Made a default implementation for archetypes and for dexterity. There is no need to do anything in `ftw.file`